src/server/DeviceServer_v2.js

```diff
@@ -145,25 +167,112 @@ class DeviceServer {
     //
     //  Wait for the keys to be ready, then start accepting connections
     //
-    const serverConfig = {
-      port: this._config.port,
-    };
+    const serverPort = this._config.port;
     server.listen(
-      serverConfig,
-      () => logger.log('Server started', serverConfig),
+      serverPort,
+      (): void => logger.log(`Server started on port: ${serverPort}`),
     );
   }
 
-  publishSpecialEvent(eventName: string, data: string, coreId: string): void {
-    this._eventPublisher.publish(
-      /* isPublic */ false,
-      eventName,
-      /* userId */ null,
+  _onDeviceSentMessage = async (
+    message: Message,
+    isPublic: boolean,
+    device: SparkCore,
+  ): Promise<void> => {
+    const deviceID = device.getHexCoreID();
+    const eventData = {
+      data: message.getPayloadLength() === 0 ? null : message.getPayload().toString(),
+      deviceID,
+      isPublic,
+      name: message.getUriPath().substr(3),
+      ttl: message.getMaxAge() > 0 ? message.getMaxAge() : 60,
+    };
+
+
+    const lowerEventName = eventData.name.toLowerCase();
```

Comment:
I almost didn't touch the filtering for `spark/` events, switched from this.setAttributes to repo methods but, there is still much work here.